src/server/frontend_xwayland/xwayland_server.cpp

```diff
@@ -44,6 +44,22 @@ namespace mf = mir::frontend;
 namespace md = mir::dispatch;
 using namespace std::chrono_literals;
 
+namespace
+{
+auto make_readable_fd(mir::Fd fd, std::function<void()> const& on_readable)
```

Comment:
Maybe the name of this function should mention that it doesn't necessarily make a `ReadableFd`?

`make_dispatcher_if_valid_fd`?

Reply:
Removed need for function